src/routers/group.ts

```diff
@@ -0,0 +1,91 @@
+import express from 'express';
+import { GroupService } from '../services/groupService';
+import {
+  groupBodyValidatorOnCreate,
+  groupBodyValidatorOnUpdate,
+  groupRelationsBodyValidator,
+  paramsIdValidator,
+} from '../validation/validators';
+import { Group } from '../models/typeORMModels';
+
+const group = express.Router();
+const groupService = new GroupService();
+
+group.post('/group', groupBodyValidatorOnCreate, async (req, res) => {
+  const group: Group = { ...req.body };
+  const result = await groupService.createGroup(group);
+  if (result instanceof Error) {
+    res.status(500).json({ message: result });
+  } else {
+    res.status(200).json({ createdGroup: result });
+  }
+});
+
+group.get('/group/:id', paramsIdValidator, async (req, res) => {
+  const { id } = req.params;
+  const result = await groupService.getGroupById(id);
+  if (result instanceof Error) {
+    res.status(500).json({ message: result });
+  } else {
+    result
+      ? res.status(200).json({ group: result })
+      : res.status(404).json({ message: `no groups with id ${id}` });
+  }
+});
+
+group.get('/groups', async (req, res) => {
+  const result = await groupService.getGroups();
+
+  if (result instanceof Error) {
+    res.status(500).json({ message: result });
+  } else {
+    res.status(200).json({ groups: result });
+  }
+});
+
+group.put(
+  '/group/:id',
+  groupBodyValidatorOnUpdate,
+  paramsIdValidator,
+  async (req, res) => {
+    const { id } = req.params;
+    const result = await groupService.updateGroup(id, req.body);
+    if (result instanceof Error) {
+      res.status(500).json({ message: result });
+    } else {
+      result
+        ? res.status(200).json({ updatedGroup: result })
+        : res.status(404).json({ message: `no groups with id ${id}` });
+    }
+  }
+);
+
+group.delete('/group/:id', paramsIdValidator, async (req, res) => {
+  const { id } = req.params;
+  const result = await groupService.deleteGroup(id);
+  if (result instanceof Error) {
+    res.status(500).json({ message: result });
+  } else {
+    result
+      ? res.status(200).json({ message: `group with id ${id} deleted` })
+      : res.status(404).json({ message: `no groups with id ${id}` });
+  }
+});
+
+group.post(
+    '/group_relations/:id',
```

Comment:
Don't you think "group_relations" isn't the best way to describe what users of your service will get from this endpoint? Maybe it would be better to use something like "/group/:groupId/addUsers"? Or "/add-users-to-group/:groupId"?

Reply:
fixed